tests/aws/services/stepfunctions/templates/errorhandling/statemachines/task_service_lambda_invoke_catch_all_output_path.json5

```diff
@@ -0,0 +1,36 @@
+{
+  "Comment": "TASK_SERVICE_LAMBDA_INVOKE_CATCH_ALL_OUTPUT_PATH",
+  "StartAt": "InvokeLambda",
+  "States": {
+    "InvokeLambda": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "Parameters": {
+        "FunctionName.$": "$.FunctionName",
+        "Payload.$": "$.Payload",
+      },
+      "ResultPath": "$.Payload",
+      "OutputPath": "$.Payload",
+      "Catch": [
+      {
+          "ErrorEquals": [
+            "States.ALL"
+          ],
+          "Next": "HandleGeneralError"
+        }
+      ],
+      "Next": "ProcessResult"
+    },
+    "ProcessResult": {
+      "Type": "Pass",
+      "End": true
+    },
+    "HandleGeneralError": {
+      "Type": "Pass",
+      "Parameters": {
+        "InputValue.$": "$"
+      },
+      "End": true
```

Comment:
I stand corrected but shouldn't we be trying to test that the reduction `"OutputPath": "$.Payload"` gets ignored here since (as per that original GitHub issue)?

```json
...
      "Catch": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "Next": "OnError",
          "ResultPath": "$.error"
        }
      ]
    },
    "OnError": {
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "OutputPath": "$.Payload", #
...
```

Am aware that the above `Catch` allows the error and input to be preserved (from https://docs.aws.amazon.com/step-functions/latest/dg/input-output-resultpath.html#input-output-resultpath-catch)

Reply:
The test checks exactly for this. Previously the computation would fail as it tried to apply the `"OutputPath"` reduction on the error result (which had no "$.Payload" path). With these changes we can see how the evaluation instead continues correctly giving as output the expected output object `{"error"..., "cause"....}` and does not apply the reductions. We then have other tests in the test suite that instead check `*Path` operate as expected in other scenarios (e.g. with no errors)

Reply:
Ahh gotcha. That makes sense since. So we would expect the above to successfully process the reduction in an error case while ignoring the `{"error"..., "cause"....}` fields since that doesn't fall apart of teh payload we are wanting to access.